[BUG] [onedir] cannot install mysqlclient pip package

**Description**
I cannot install the `mysqlclient` package into the tiamat's `/opt/saltstack/salt/pypath`. That is e.g. necessary for the `mysql` module and `mysql_*` states to work.

**Steps to Reproduce the behavior**

* Install salt 3005 (onedir) on Ubuntu 20.04
* Install build dependencies `apt-get install default-libmysqlclient-dev build-essential` (`python3-dev` omitted since salt's python is used; installing `python3-dev` anyways doesn't make a difference – as expected)
* `salt pip install mysqlclient`
```
Collecting mysqlclient
  Using cached mysqlclient-2.1.1.tar.gz (88 kB)
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: mysqlclient
  Building wheel for mysqlclient (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /opt/saltstack/salt/run/run -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-3_ih_2ow/mysqlclient_e36916713b80449e8d196092277be980/setup.py'"'"'; __file__='"'"'/tmp/pip-install-3_ih_2ow/mysqlclient_e36916713b80449e8d196092277be980/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-ftqymcpb
       cwd: /tmp/pip-install-3_ih_2ow/mysqlclient_e36916713b80449e8d196092277be980/
  Complete output (46 lines):
  mysql_config --version
  ['8.0.30']
  mysql_config --libs
  ['-L/usr/lib/x86_64-linux-gnu', '-lmysqlclient', '-lpthread', '-ldl', '-lssl', '-lcrypto', '-lresolv', '-lm', '-lrt']
  mysql_config --cflags
  ['-I/usr/include/mysql']
  ext_options:
    library_dirs: ['/usr/lib/x86_64-linux-gnu']
    libraries: ['mysqlclient', 'pthread', 'dl', 'resolv', 'm', 'rt']
    extra_compile_args: ['-std=c99']
    extra_link_args: []
    include_dirs: ['/usr/include/mysql']
    extra_objects: []
    define_macros: [('version_info', "(2,1,1,'final',0)"), ('__version__', '2.1.1')]
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.9
  creating build/lib.linux-x86_64-3.9/MySQLdb
  copying MySQLdb/__init__.py -> build/lib.linux-x86_64-3.9/MySQLdb
  copying MySQLdb/_exceptions.py -> build/lib.linux-x86_64-3.9/MySQLdb
  copying MySQLdb/connections.py -> build/lib.linux-x86_64-3.9/MySQLdb
  copying MySQLdb/converters.py -> build/lib.linux-x86_64-3.9/MySQLdb
  copying MySQLdb/cursors.py -> build/lib.linux-x86_64-3.9/MySQLdb
  copying MySQLdb/release.py -> build/lib.linux-x86_64-3.9/MySQLdb
  copying MySQLdb/times.py -> build/lib.linux-x86_64-3.9/MySQLdb
  creating build/lib.linux-x86_64-3.9/MySQLdb/constants
  copying MySQLdb/constants/__init__.py -> build/lib.linux-x86_64-3.9/MySQLdb/constants
  copying MySQLdb/constants/CLIENT.py -> build/lib.linux-x86_64-3.9/MySQLdb/constants
  copying MySQLdb/constants/CR.py -> build/lib.linux-x86_64-3.9/MySQLdb/constants
  copying MySQLdb/constants/ER.py -> build/lib.linux-x86_64-3.9/MySQLdb/constants
  copying MySQLdb/constants/FIELD_TYPE.py -> build/lib.linux-x86_64-3.9/MySQLdb/constants
  copying MySQLdb/constants/FLAG.py -> build/lib.linux-x86_64-3.9/MySQLdb/constants
  warning: build_py: byte-compiling is disabled, skipping.
  
  running build_ext
  building 'MySQLdb._mysql' extension
  creating build/temp.linux-x86_64-3.9
  creating build/temp.linux-x86_64-3.9/MySQLdb
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -g -O2 -fdebug-prefix-map=/tmp/tmpv2n0pvhl/salt_3005+ds=. -fstack-protector-strong -Wformat -Werror=format-security -g -O2 -fdebug-prefix-map=/tmp/tmpv2n0pvhl/salt_3005+ds=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -Dversion_info=(2,1,1,'final',0) -D__version__=2.1.1 -I/usr/include/mysql -I/opt/saltstack/salt/run/include/python3.9 -c MySQLdb/_mysql.c -o build/temp.linux-x86_64-3.9/MySQLdb/_mysql.o -std=c99
  MySQLdb/_mysql.c:46:10: fatal error: Python.h: No such file or directory
     46 | #include "Python.h"
        |          ^~~~~~~~~~
  compilation terminated.
  error: command '/usr/bin/gcc' failed with exit code 1
  ----------------------------------------
  ERROR: Failed building wheel for mysqlclient
  Running setup.py clean for mysqlclient
Failed to build mysqlclient
Installing collected packages: mysqlclient
    Running setup.py install for mysqlclient ... error
    ERROR: Command errored out with exit status 1:
     command: /opt/saltstack/salt/run/run -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-3_ih_2ow/mysqlclient_e36916713b80449e8d196092277be980/setup.py'"'"'; __file__='"'"'/tmp/pip-install-3_ih_2ow/mysqlclient_e36916713b80449e8d196092277be980/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-ff0b826_/install-record.txt --single-version-externally-managed --compile --install-headers /opt/saltstack/salt/run/include
         cwd: /tmp/pip-install-3_ih_2ow/mysqlclient_e36916713b80449e8d196092277be980/
    Complete output (48 lines):
    mysql_config --version
    ['8.0.30']
    mysql_config --libs
    ['-L/usr/lib/x86_64-linux-gnu', '-lmysqlclient', '-lpthread', '-ldl', '-lssl', '-lcrypto', '-lresolv', '-lm', '-lrt']
    mysql_config --cflags
    ['-I/usr/include/mysql']
    ext_options:
      library_dirs: ['/usr/lib/x86_64-linux-gnu']
      libraries: ['mysqlclient', 'pthread', 'dl', 'resolv', 'm', 'rt']
      extra_compile_args: ['-std=c99']
      extra_link_args: []
      include_dirs: ['/usr/include/mysql']
      extra_objects: []
      define_macros: [('version_info', "(2,1,1,'final',0)"), ('__version__', '2.1.1')]
    running install
    setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
      warnings.warn(
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.9
    creating build/lib.linux-x86_64-3.9/MySQLdb
    copying MySQLdb/__init__.py -> build/lib.linux-x86_64-3.9/MySQLdb
    copying MySQLdb/_exceptions.py -> build/lib.linux-x86_64-3.9/MySQLdb
    copying MySQLdb/connections.py -> build/lib.linux-x86_64-3.9/MySQLdb
    copying MySQLdb/converters.py -> build/lib.linux-x86_64-3.9/MySQLdb
    copying MySQLdb/cursors.py -> build/lib.linux-x86_64-3.9/MySQLdb
    copying MySQLdb/release.py -> build/lib.linux-x86_64-3.9/MySQLdb
    copying MySQLdb/times.py -> build/lib.linux-x86_64-3.9/MySQLdb
    creating build/lib.linux-x86_64-3.9/MySQLdb/constants
    copying MySQLdb/constants/__init__.py -> build/lib.linux-x86_64-3.9/MySQLdb/constants
    copying MySQLdb/constants/CLIENT.py -> build/lib.linux-x86_64-3.9/MySQLdb/constants
    copying MySQLdb/constants/CR.py -> build/lib.linux-x86_64-3.9/MySQLdb/constants
    copying MySQLdb/constants/ER.py -> build/lib.linux-x86_64-3.9/MySQLdb/constants
    copying MySQLdb/constants/FIELD_TYPE.py -> build/lib.linux-x86_64-3.9/MySQLdb/constants
    copying MySQLdb/constants/FLAG.py -> build/lib.linux-x86_64-3.9/MySQLdb/constants
    warning: build_py: byte-compiling is disabled, skipping.
    
    running build_ext
    building 'MySQLdb._mysql' extension
    creating build/temp.linux-x86_64-3.9
    creating build/temp.linux-x86_64-3.9/MySQLdb
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -g -O2 -fdebug-prefix-map=/tmp/tmpv2n0pvhl/salt_3005+ds=. -fstack-protector-strong -Wformat -Werror=format-security -g -O2 -fdebug-prefix-map=/tmp/tmpv2n0pvhl/salt_3005+ds=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -Dversion_info=(2,1,1,'final',0) -D__version__=2.1.1 -I/usr/include/mysql -I/opt/saltstack/salt/run/include/python3.9 -c MySQLdb/_mysql.c -o build/temp.linux-x86_64-3.9/MySQLdb/_mysql.o -std=c99
    MySQLdb/_mysql.c:46:10: fatal error: Python.h: No such file or directory
       46 | #include "Python.h"
          |          ^~~~~~~~~~
    compilation terminated.
    error: command '/usr/bin/gcc' failed with exit code 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /opt/saltstack/salt/run/run -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-3_ih_2ow/mysqlclient_e36916713b80449e8d196092277be980/setup.py'"'"'; __file__='"'"'/tmp/pip-install-3_ih_2ow/mysqlclient_e36916713b80449e8d196092277be980/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-ff0b826_/install-record.txt --single-version-externally-managed --compile --install-headers /opt/saltstack/salt/run/include Check the logs for full command output.
```
* Execute `salt pip install --global-option=build_ext --global-option="-I/opt/saltstack/salt/run/include/python/" mysqlclient` (which should not be necessary)
```
pip/_internal/commands/install.py:245: UserWarning: Disabling all use of wheels due to the use of --build-option / --global-option / --install-option.
  cmdoptions.check_install_build_global(options)
Collecting mysqlclient
  Using cached mysqlclient-2.1.1.tar.gz (88 kB)
  Preparing metadata (setup.py) ... done
Skipping wheel build for mysqlclient, due to binaries being disabled for it.
Installing collected packages: mysqlclient
    Running setup.py install for mysqlclient ... error
    ERROR: Command errored out with exit status 1:
     command: /opt/saltstack/salt/run/run -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-nbinds0t/mysqlclient_f2a9decf59c84e928a89c422310358d9/setup.py'"'"'; __file__='"'"'/tmp/pip-install-nbinds0t/mysqlclient_f2a9decf59c84e928a89c422310358d9/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' build_ext -I/opt/saltstack/salt/run/include/python/ install --record /tmp/pip-record-tuw4zl42/install-record.txt --single-version-externally-managed --compile --install-headers /opt/saltstack/salt/run/include
         cwd: /tmp/pip-install-nbinds0t/mysqlclient_f2a9decf59c84e928a89c422310358d9/
    Complete output (39 lines):
    mysql_config --version
    ['8.0.30']
    mysql_config --libs
    ['-L/usr/lib/x86_64-linux-gnu', '-lmysqlclient', '-lpthread', '-ldl', '-lssl', '-lcrypto', '-lresolv', '-lm', '-lrt']
    mysql_config --cflags
    ['-I/usr/include/mysql']
    ext_options:
      library_dirs: ['/usr/lib/x86_64-linux-gnu']
      libraries: ['mysqlclient', 'pthread', 'dl', 'resolv', 'm', 'rt']
      extra_compile_args: ['-std=c99']
      extra_link_args: []
      include_dirs: ['/usr/include/mysql']
      extra_objects: []
      define_macros: [('version_info', "(2,1,1,'final',0)"), ('__version__', '2.1.1')]
    running build_ext
    building 'MySQLdb._mysql' extension
    creating build
    creating build/temp.linux-x86_64-3.9
    creating build/temp.linux-x86_64-3.9/MySQLdb
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -g -O2 -fdebug-prefix-map=/tmp/tmpv2n0pvhl/salt_3005+ds=. -fstack-protector-strong -Wformat -Werror=format-security -g -O2 -fdebug-prefix-map=/tmp/tmpv2n0pvhl/salt_3005+ds=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -Dversion_info=(2,1,1,'final',0) -D__version__=2.1.1 -I/usr/include/mysql -I/opt/saltstack/salt/run/include/python/ -I/opt/saltstack/salt/run/include/python3.9 -c MySQLdb/_mysql.c -o build/temp.linux-x86_64-3.9/MySQLdb/_mysql.o -std=c99
    In file included from /opt/saltstack/salt/run/include/python/Python.h:8,
                     from MySQLdb/_mysql.c:46:
    /opt/saltstack/salt/run/include/python/pyconfig.h:1619: warning: "_POSIX_C_SOURCE" redefined
     1619 | #define _POSIX_C_SOURCE 200809L
          |
    In file included from /usr/include/x86_64-linux-gnu/bits/libc-header-start.h:33,
                     from /usr/include/stdint.h:26,
                     from /usr/lib/gcc/x86_64-linux-gnu/9/include/stdint.h:9,
                     from /usr/include/mysql/mysql.h:45,
                     from MySQLdb/_mysql.c:29:
    /usr/include/features.h:310: note: this is the location of the previous definition
      310 | # define _POSIX_C_SOURCE 199506L
          |
    creating build/lib.linux-x86_64-3.9
    creating build/lib.linux-x86_64-3.9/MySQLdb
    gcc -pthread -shared -L/root/.pyenv/versions/3.9.13/lib -Wl,-rpath=/root/.pyenv/versions/3.9.13/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -L/root/.pyenv/versions/3.9.13/lib -Wl,-rpath=/root/.pyenv/versions/3.9.13/lib -Wl,-Bsymbolic-functions -Wl,-z,relro build/temp.linux-x86_64-3.9/MySQLdb/_mysql.o -L/usr/lib/x86_64-linux-gnu -L/root/.pyenv/versions/3.9.13/lib -lmysqlclient -lpthread -ldl -lresolv -lm -lrt -o build/lib.linux-x86_64-3.9/MySQLdb/_mysql.cpython-39-x86_64-linux-gnu.so
    collect2: fatal error: cannot find 'ld'
    compilation terminated.
    error: command '/usr/bin/gcc' failed with exit code 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /opt/saltstack/salt/run/run -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-nbinds0t/mysqlclient_f2a9decf59c84e928a89c422310358d9/setup.py'"'"'; __file__='"'"'/tmp/pip-install-nbinds0t/mysqlclient_f2a9decf59c84e928a89c422310358d9/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' build_ext -I/opt/saltstack/salt/run/include/python/ install --record /tmp/pip-record-tuw4zl42/install-record.txt --single-version-externally-managed --compile --install-headers /opt/saltstack/salt/run/include Check the logs for full command output.
Distribution mysqlclient was not found installed
```

**Expected behavior**
No error message. `salt pip install mysqlclient` should just work.

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```yaml
Salt Version:
          Salt: 3005
 
Dependency Versions:
          cffi: 1.14.6
      cherrypy: unknown
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.1.0
       libgit2: Not Installed
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: 3.9.8
        pygit2: Not Installed
        Python: 3.9.13 (main, Aug 23 2022, 18:31:04)
  python-gnupg: 0.4.8
        PyYAML: 5.4.1
         PyZMQ: 23.2.0
         smmap: Not Installed
       timelib: 0.2.4
       Tornado: 4.5.3
           ZMQ: 4.3.4
 
System Versions:
          dist: ubuntu 20.04 focal
        locale: utf-8
       machine: x86_64
       release: 5.4.0-125-generic
        system: Linux
       version: Ubuntu 20.04 focal
```
</details>


Sorry if this is the wrong repository for this bug report. Please advice me on the right one if that is the case.

--- a/changelog/62567.fixed
+++ b/changelog/62567.fixed
@@ -1,0 +1,1 @@
+Fix the Pyinstaller hooks to preserve the environment if None is passed.

--- a/changelog/62628.fixed
+++ b/changelog/62628.fixed
@@ -1,0 +1,1 @@
+Fix the Pyinstaller hooks to preserve the environment if None is passed.

--- a/salt/utils/pyinstaller/rthooks/_overrides.py
+++ b/salt/utils/pyinstaller/rthooks/_overrides.py
@@ -17,7 +17,7 @@ def clean_pyinstaller_vars(environ):
     Restore or cleanup PyInstaller specific environent variable behavior.
     """
     if environ is None:
-        environ = {}
+        environ = os.environ.copy()
     # When Salt is bundled with tiamat, it MUST NOT contain LD_LIBRARY_PATH
     # when shelling out, or, at least the value of LD_LIBRARY_PATH set by
     # pyinstaller.

--- a/salt/utils/pyinstaller/rthooks/_overrides.py
+++ b/salt/utils/pyinstaller/rthooks/_overrides.py
@@ -17,7 +17,7 @@ def clean_pyinstaller_vars(environ):
     Restore or cleanup PyInstaller specific environent variable behavior.
     """
     if environ is None:
-        environ = os.environ.copy()
+        environ = dict(os.environ)
     # When Salt is bundled with tiamat, it MUST NOT contain LD_LIBRARY_PATH
     # when shelling out, or, at least the value of LD_LIBRARY_PATH set by
     # pyinstaller.

--- a/tests/pytests/functional/utils/pyinstaller/rthooks/test_salt_utils_vt_terminal.py
+++ b/tests/pytests/functional/utils/pyinstaller/rthooks/test_salt_utils_vt_terminal.py
@@ -16,8 +16,8 @@ def envvar(request):
 @pytest.fixture
 def meipass(envvar):
     with mock.patch("salt.utils.pyinstaller.rthooks._overrides.sys") as patched_sys:
-        patched_sys._MEIPASS = envvar
-        assert overrides.sys._MEIPASS == envvar
+        patched_sys._MEIPASS = "{}_VALUE".format(envvar)
+        assert overrides.sys._MEIPASS == "{}_VALUE".format(envvar)
         yield "{}_VALUE".format(envvar)
     assert not hasattr(sys, "_MEIPASS")
     assert not hasattr(overrides.sys, "_MEIPASS")
@@ -94,7 +94,6 @@ def test_vt_terminal_environ_cleanup(envvar, meipass):
                 "-c",
                 "import os, json; print(json.dumps(dict(os.environ)))",
             ],
-            env={},
             stream_stdout=False,
             stream_stderr=False,
         )
@@ -116,7 +115,7 @@ def test_vt_terminal_environ_cleanup(envvar, meipass):
 
 def test_vt_terminal_environ_cleanup_passed_directly_not_removed(envvar, meipass):
     env = {
-        envvar: meipass,
+        envvar: envvar,
     }
     original_env = dict(os.environ)
 
@@ -139,4 +138,4 @@ def test_vt_terminal_environ_cleanup_passed_directly_not_removed(envvar, meipass
     returned_env = json.loads(buffer_o)
     assert returned_env != original_env
     assert envvar in returned_env
-    assert returned_env[envvar] == meipass
+    assert returned_env[envvar] == envvar

--- a/tests/pytests/functional/utils/pyinstaller/rthooks/test_salt_utils_vt_terminal.py
+++ b/tests/pytests/functional/utils/pyinstaller/rthooks/test_salt_utils_vt_terminal.py
@@ -94,6 +94,7 @@ def test_vt_terminal_environ_cleanup(envvar, meipass):
                 "-c",
                 "import os, json; print(json.dumps(dict(os.environ)))",
             ],
+            env={},
             stream_stdout=False,
             stream_stderr=False,
         )

--- a/tests/pytests/functional/utils/pyinstaller/rthooks/test_subprocess.py
+++ b/tests/pytests/functional/utils/pyinstaller/rthooks/test_subprocess.py
@@ -17,8 +17,8 @@ def envvar(request):
 @pytest.fixture
 def meipass(envvar):
     with mock.patch("salt.utils.pyinstaller.rthooks._overrides.sys") as patched_sys:
-        patched_sys._MEIPASS = envvar
-        assert overrides.sys._MEIPASS == envvar
+        patched_sys._MEIPASS = "{}_VALUE".format(envvar)
+        assert overrides.sys._MEIPASS == "{}_VALUE".format(envvar)
         yield "{}_VALUE".format(envvar)
     assert not hasattr(sys, "_MEIPASS")
     assert not hasattr(overrides.sys, "_MEIPASS")
@@ -92,7 +92,7 @@ def test_subprocess_popen_environ_cleanup(envvar, meipass):
 
 def test_subprocess_popen_environ_cleanup_passed_directly_not_removed(envvar, meipass):
     env = {
-        envvar: meipass,
+        envvar: envvar,
     }
     original_env = dict(os.environ)
 
@@ -107,4 +107,4 @@ def test_subprocess_popen_environ_cleanup_passed_directly_not_removed(envvar, me
     returned_env = json.loads(stdout)
     assert returned_env != original_env
     assert envvar in returned_env
-    assert returned_env[envvar] == meipass
+    assert returned_env[envvar] == envvar

